File: backend-springboot/src/main/java/br/com/portalproposta/backend/controller/ProposalTypesController.java
package br.com.portalproposta.backend.controller;

import br.com.portalproposta.backend.model.TipoProposta;
import br.com.portalproposta.backend.service.TipoPropostaService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

import java.net.URI;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/proposal-types")
public class ProposalTypesController {

    private final TipoPropostaService proposalTypeService;

    @Autowired
    public ProposalTypesController(TipoPropostaService proposalTypeService) {
        this.proposalTypeService = proposalTypeService;
    }

    @GetMapping
    public ResponseEntity<List<TipoProposta>> list() {
        return ResponseEntity.ok(proposalTypeService.list());
    }

    @PostMapping
    public ResponseEntity<TipoProposta> create(@RequestBody TipoProposta tipo) {
        TipoProposta created = proposalTypeService.create(tipo);
        URI uri = ServletUriComponentsBuilder.fromCurrentRequest()
                .path("/{id}")
                .buildAndExpand(created.getId())
                .toUri();
        return ResponseEntity.created(uri).body(created);
    }

    @GetMapping("/{id}")
    public ResponseEntity<TipoProposta> get(@PathVariable String id) {
        return ResponseEntity.ok(proposalTypeService.getById(id));
    }

    @PutMapping("/{id}")
    public ResponseEntity<TipoProposta> update(@PathVariable String id, @RequestBody TipoProposta tipo) {
        TipoProposta updated = proposalTypeService.update(id, tipo);
        return ResponseEntity.ok(updated);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Map<String, String>> delete(@PathVariable String id) {
        proposalTypeService.delete(id);
        Map<String, String> response = new HashMap<>();
        response.put("message", "Tipo de proposta excluído com sucesso");
        return ResponseEntity.ok(response);
    }
}

File: backend/Controllers/ProposalTypesController.cs
using Microsoft.AspNetCore.Mvc;
using PortalProposta.Models;
using PortalProposta.Services;

namespace PortalProposta.Controllers;

[ApiController]
[Route("api/proposal-types")]
public class ProposalTypesController(TipoPropostaService proposalTypeService) : ControllerBase
{
    [HttpGet]
    public async Task<ActionResult<List<TipoProposta>>> List() =>
        Ok(await proposalTypeService.ListAsync());

    [HttpPost]
    public async Task<ActionResult<TipoProposta>> Create(TipoProposta tipo) =>
        CreatedAtAction(nameof(Get), new { id = (await proposalTypeService.CreateAsync(tipo)).Id }, tipo);

    [HttpGet("{id}")]
    public async Task<ActionResult<TipoProposta>> Get(string id) =>
        Ok(await proposalTypeService.GetByIdAsync(id));

    [HttpPut("{id}")]
    public async Task<ActionResult<TipoProposta>> Update(string id, TipoProposta tipo) =>
        Ok(await proposalTypeService.UpdateAsync(id, tipo));

    [HttpDelete("{id}")]
    public async Task<IActionResult> Delete(string id)
    {
        await proposalTypeService.DeleteAsync(id);
        return Ok(new { message = "Tipo de proposta excluído com sucesso" });
    }
}
